### ISSUE TITLE
Error in MulticastDelegate PlayerDisconnected

### ISSUE THREAD

**Issue by d3adwolf** [LeandroTheDev/feex_ranks_restored#1]
I add an error from the logs for reporting, so far I have not seen that it somehow interferes or breaks the functionality, but since there is a bug, we need to fix it in the future. 

The error appears when logging in and out of the server. I also thought that the problem was with RocketMod core, but we discussed it with the author of RocketModFix - the problem is on the plugin side.

If you need additional information, I'm ready to provide it.

https://github.com/RocketModFix/RocketModFix/issues/81

```csharp
2024-05-29 16:24:07.035	
  at Rocket.Core.Extensions.MulticastDelegateExtension.TryInvoke (System.MulticastDelegate theDelegate, System.Object[] args) [0x00013] in <2b8ee4b602174810b7294320f1aecbc4>:0 
2024-05-29 16:24:07.035	
  at System.Delegate.DynamicInvoke (System.Object[] args) [0x00000] in <2161c9885dd04ff291b79db9349c4b9b>:0 
2024-05-29 16:24:07.035	
  at System.MulticastDelegate.DynamicInvokeImpl (System.Object[] args) [0x00008] in <2161c9885dd04ff291b79db9349c4b9b>:0 
2024-05-29 16:24:07.035	
  at System.Delegate.DynamicInvokeImpl (System.Object[] args) [0x000e7] in <2161c9885dd04ff291b79db9349c4b9b>:0 
2024-05-29 16:24:07.035	
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <2161c9885dd04ff291b79db9349c4b9b>:0 
2024-05-29 16:24:07.035	
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00083] in <2161c9885dd04ff291b79db9349c4b9b>:0 
2024-05-29 16:24:07.035	
   --- End of inner exception stack trace ---
2024-05-29 16:24:07.035	
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in <2161c9885dd04ff291b79db9349c4b9b>:0 
2024-05-29 16:24:07.035	
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
2024-05-29 16:24:07.035	
  at FeexRanks.FeexRanksPlugin.OnPlayerDisconnected (Rocket.Unturned.Player.UnturnedPlayer player) [0x00049] in <9bd3ca080dab4fcbbb1be2b12821fe32>:0 
2024-05-29 16:24:07.035	
[5/29/2024 1:24:06 PM] [Error] Error in MulticastDelegate PlayerDisconnected: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object
```

**Comment by sunnamed434** [comment]
The problem is plugin could be unloaded at some point and OnPlayerDisconnected event is not unsubscribed anywhere (and after many reloads of the plugin event listeners will stack, stack and stack every reload which will cause a lot more troubles) in the code this is problem N1 and the problem which can actually cause this problem

**Comment by LeandroTheDev** [comment]
I think this is caused because the configuration PointsEarnPerTime is 0 and this will not create the tickrate instance
![image](https://github.com/user-attachments/assets/bf4fe04a-a468-4463-b2e7-a263db23ab09)

And the OnPlayerDisconnected functions always tries to call the tickate.RemovePlayer()

![image](https://github.com/user-attachments/assets/a56f0751-3a09-44a3-b43a-9219792066dc)

Giving he null exception, i will make a handle for tickrate null, and check if this fixes the problem: [75839e1](https://github.com/LeandroTheDev/feex_ranks_restored/commit/75839e11785ac7355b9c01a241f3fc8649eb2177)
